### PR TITLE
Optimize handle function

### PR DESCRIPTION
**Pull request by gnought**

### CODE REVIEW

File: test/handlers/connect.js [hunk 1]
```diff
@@ -13,12 +13,16 @@ test('reject clients with no clientId running on MQTT 3.1', function (t) {
   }
 
   client.broker = broker
+  client.conn = {
+    destroy: function () {}
```
Comment (mcollina) [inline test/handlers/connect.js]: Why this change?

Comment (gnought) [reply test/handlers/connect.js]: `handleConnect` is a sub handler in `handle`, and there's part of `connect` login in `handle` function. It is best to emulate much more real world case.

Comment (mcollina) [reply test/handlers/connect.js]: If this is called, would you mind placing an assertion in there?

File: test/handlers/connect.js [hunk 2]
```diff
@@ -13,12 +14,16 @@ test('reject clients with no clientId running on MQTT 3.1', function (t) {
   }
 
   client.broker = broker
+  client.conn = {
+    destroy: function () { assert.fail('should not destroy') }
```
Comment (mcollina) [inline test/handlers/connect.js]: I think you can use t.fail() here instead
